Install securedrop-log and setup

kushaldas opened the pull request:
## Status

Ready for review 

## Description of Changes

Resolves #440 

Sets up `securedrop-log` service in related VMs. All logs should be flowing into `sd-log` vm.

Changes proposed in this pull request:

## Testing

- [ ] `make all` should have logging ready.
- [ ] Checks logs at `/home/user/QubesIncomingLogs/` directory in `sd-logs` vm

## Checklist

### If you have made changes to the provisioning logic

- [ ] Linting (`make flake8`) and tests (`make test`) pass in dom0 of a Qubes install

- [ ] I have added/removed files, and have updated packaging logic in MANIFEST.in and rpm-build/SPECS/securedrop-workstation-dom0-config.spec

emkll commented on Makefile:
```diff
@@ -62,7 +62,7 @@ sd-app: prep-salt ## Provisions SD APP VM
 sd-whonix: prep-salt ## Provisions SD Whonix VM
 	sudo qubesctl --show-output state.sls sd-whonix
-	sudo qubesctl --show-output --skip-dom0 --targets sd-whonix-buster-template,sd-whonix state.highstate
+	sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15,sd-whonix state.highstate
```
Why was this changed? sd-whonix-buster-template is cloned off of whonix-gw-15

kushaldas replied:
I don't have a `sd-whonix-buster-template`. What am i missing?

emkll replied:
Ahh, right, logging in `sd-whonix`. Good catch, `sd-whonix-buster-template` doesn't exist so it's effectively a no-op here. What will happen to `sys-whonix` ? The package will be installed but since the configuration is not in place, nothing will happen there?

emkll commented on dom0/sd-workstation.top:
```diff
@@ -9,14 +9,19 @@ base:
     - sd-remove-unused-templates
-    - sd-log
+
+  sd-log-buster-template:
```
i don't think the order here is important, it depends what the `requires` state

kushaldas replied:
Oh, that is something I was wondering.

emkll commented on dom0/sd-workstation.top:
```diff
@@ -33,12 +38,12 @@ base:
     - sd-workstation-template-files
   sys-usb:
     - sd-usb-autoattach-add
+  whonix-gw-15:
```
Why do we apply sd-whonix-template files to whonix-gw-15? shouldn't we apply it to `sd-whonix`?

kushaldas replied:
@emkll I can not find any `sd-whonix-template`, the `sd-whonix` uses `whonix-gw-15` and that is why I am applying files to that template.

emkll commented on dom0/sd-workstation-template-files.sls:
```diff
@@ -11,6 +11,14 @@ sd-workstation-template-install-kernel-config-packages:
       - sls: fpf-apt-test-repo
 
+
+sd-workstation-install-securedrop-log-package:
```
All other templates are cloned from this template, but in an in-place scenario, the template would not get upgraded. One would need to make clean and make all for these changes to propagate to all the other sd-* templates. I think the approach makes sense here, install in the base template, and also install it vm-specific states

emkll commented on dom0/sd-app-files.sls:
```diff
@@ -18,3 +18,18 @@ install-securedrop-client-package:
+    - require:
+      - sls: fpf-apt-test-repo
+
+sd-rsyslog-for-sd-app:
```
Do we want this config to be there before or after installing the package? Do we need to restart the service afterwards?

kushaldas replied:
either will do, I am enabling the service afterwards.

emkll commented on dom0/sd-app-files.sls:
```diff
@@ -18,3 +18,18 @@ install-securedrop-client-package:
+install-securedrop-log-package:
+  pkg.installed:
+    - pkgs:
+      - securedrop-log
```
It might be easier to read by adding this as a list item to the previous task

conorsch replied:
Definitely fold into lists where possible. In salt, all individual tasks _must_ be uniquely named, whereas in this PR the `install-securedrop-log-package` task name is duplicated in several locations. If that's not throwing an error, we probably got lucky in that the states aren't combined for a single run... still, needs to be cleaned up.

emkll commented on dom0/sd-rsyslog.conf.j2:
```diff
@@ -0,0 +1,3 @@
+[sd-rsyslog]
```
:+1: templating

conorsch replied:
Templating definitely good. We might be able to use `grains["localhost"]` to get the VM name, rather than passing in `context` every time. (See output of e.g. `sudo qubesctl --skip-dom0 --show-output --target sys-firewall grains.items`.) If that works, we could stick the tasks for 1) install pkg; 2) configure rsyslog in a single state file and include it, reducing a lot of duplication.

emkll commented on dom0/sd-log-reboot.sls:
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
```
Is there a reason you've put this in a separate file? Because there's to `requires` here, this means that `sd-log` will be rebooted at a time that is likely not the time you are expecting here

kushaldas replied:
Yes, this is where I need help, how to make sure that we reboot `sd-log` in proper time, may be via an instruction in documentation and remove this totally.

kushaldas commented:
>If the goal is to bootstrap logging as early as possible in the provisioning phase (to get some provisioning logs in sd-logs as well), it might make sense to apply the state to dom0 and then to sd-log early in the process, as part of a separate qubesctl call, similar to what was done in 

Okay, trying to get this in.

kushaldas commented:
We need to get this merged https://github.com/freedomofpress/securedrop-log/pull/12

emkll commented on scripts/provision-all:
```diff
@@ -17,6 +17,11 @@ sudo qubesctl --show-output --skip-dom0 --targets sys-firewall state.sls sd-sys-
 sudo qubesctl --show-output state.highstate
 
+echo "Setup sd-log vm first"
+sudo qubesctl --show-output --skip-dom0 --targets sd-log state.sls sd-log-templates-files
```
should be `sd-log-template-files` here

emkll replied:
I think the issue here is more than a typo (this is untested, merely suggesting):
```
sudo qubesctl --show-output --skip-dom0 --targets sd-log-template, sd-log state.highstate
```

I am getting an error after make clean that sd-log can't download the securedrop-log package (which makes sense because sd-log doesn't have internet connectivity, we need to apply state to `sd-log-buster-template`

emkll commented on dom0/sd-whonix-template-files.sls:
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
```
Since this is VMs that are not deleted in place, we should remove this configuration as part of `make clean`. I suggest we implement this logic in sd-clean-all https://github.com/freedomofpress/securedrop-workstation/blob/configure_logging_world/dom0/sd-clean-all.sls

emkll commented on scripts/provision-all:
```diff
@@ -17,6 +17,11 @@ sudo qubesctl --show-output --skip-dom0 --targets sys-firewall state.sls sd-sys-
+echo "Setup sd-log vm first"
+sudo qubesctl --show-output --skip-dom0 --targets sd-log state.sls sd-log-templates-files
+# Make sure that the VM is in shutdown state
+qvm-shutdown sd-log
```
Do we still need `sd-log-reboot.sls` with this logic here? Surely `sd-log` should reboot if a log needs to be forwarded?

emkll commented on tests/base.py:
```diff
@@ -112,3 +112,19 @@ def _fileExists(self, remote_path):
 
         return True
+
+    def logging_configured(self, vmname = ""):
```
:+1: nice tests, confirm these are being run as part of `make test`

emkll commented on dom0/sd-workstation.top:
```diff
@@ -9,14 +9,19 @@ base:
     - sd-whonix
     - sd-remove-unused-templates
-    - sd-log
+
```
nit: empty line

emkll commented on dom0/sd-workstation.top:
```diff
@@ -9,14 +9,19 @@ base:
+
+  sd-log-buster-template:
+    - sd-log-template-files
+  sd-log:
```
I don't think this is required since you are shutting down this machine after provisioning the template

emkll commented on dom0/sd-log-reboot.sls:
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
```
I don't think this is required anymore, given the `qvm-shutdown sd-log` call you make in `./scripts/provision-all`. Am I missing something?

emkll replied:
We could also run `qvm-shutdown && qvm start` if we need it to be up, instead of having to maintain this complex logic

kushaldas commented:
While testing locally, `sd-proxy-buster-template` is failing to fetch repository data.

```
W: Failed to fetch https://apt-test.freedom.press/dists/buster/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect   Server: tinyproxy/1.10.0 Content-Type: text/html Connection: close   [IP: 127.0.0.1:8082]
```

Same for `debian-security`, `buster`, `whonix` repos.

emkll commented on dom0/sd-clean-all.sls:
```diff
@@ -27,6 +27,12 @@ remove-dom0-sdw-config-files:
+sd-cleanup-whonix-gw-15:
+  cmd.run:
+    - names:
+      - qvm-run whonix-gw-15 'sudo apt remove securedrop-log'
```
you probably want to do use `apt remove -y` here, and `make clean` fails here: 
```
----------
          ID: sd-cleanup-whonix-gw-15
    Function: cmd.run
        Name: qvm-run whonix-gw-15 'sudo apt remove securedrop-log'
      Result: False
     Comment: Command "qvm-run whonix-gw-15 'sudo apt remove securedrop-log'" run
     Started: 11:28:08.536188
    Duration: 8211.254 ms
     Changes:   
              ----------
              pid:
                  10820
              retcode:
                  1
              stderr:
                  Running 'sudo apt remove securedrop-log' on whonix-gw-15
                  whonix-gw-15: command failed with code: 1
              stdout:
```

emkll commented on scripts/provision-all:
```diff
@@ -17,6 +17,11 @@ sudo qubesctl --show-output --skip-dom0 --targets sys-firewall state.sls sd-sys-
+echo "Setup sd-log-buster-template vm first"
+sudo qubesctl --show-output --skip-dom0 --targets sd-log-buster-template state.sls sd-log-template-files
+# Make sure that the VM is in shutdown state
+qvm-shutdown sd-log
```
Are you sure this is still required? When running `make all`, I see in standard out:
```
sd-log: Shutdown error: Domain is powered off: 'sd-log'
````

kushaldas commented:
@emkll pushed again.

emkll commented:
Thanks for the quick changes @kushaldas . I did another visual pass on this.  I expect the order of operations for the RPC grants when running `make clean` still an issue here. Is this what you've observed locally in your testing?
 
Perhaps requiring `sd-cleanup-whonix-gw-15` as part of the `sd-cleanup-rpm-policy-grants` step (maybe securedrop-log service needs to be stopped?) and ensuring all securedrop VMs are destroyed *before* `sd-cleanup-rpm-policy-grants` is run.  The requires syntax should look similar to what you wrote in https://github.com/freedomofpress/securedrop-workstation/pull/447/files#diff-bdc48ae124ff6ab5c5028ec4ca468853R26-R27 . 

Once the make clean issues been addressed, I propose the following test plan. Would you mind walking through it as well so we have good coverage here?

- `make clean`
- `make all` on this branch
- [ ] QubesIncomingLogs in `sd-log` contains `sd-app` logs
- [ ] QubesIncomingLogs in `sd-log` contains `sd-proxy` logs
- [ ] QubesIncomingLogs in `sd-log` contains `sd-whonix` logs
- [ ] QubesIncomingLogs in `sd-log` contains `sd-devices` logs (after exporting a file to drive or print)
- [ ] QubesIncomingLogs in `sd-log` contains `sd-viewer` logs (after opening a submission in a dispvm)
- [ ] `make clean` works as expected
- [ ] There are no RPC popups after make clean
- [ ] whonix-gw-15 no longer contains the syslog customizations

eloquence commented:
@kushaldas Small note, can you put "Fixes #xxx" type descriptions in the _body_ of a PR, instead of the title? This makes titles a bit easier to read, and ensures that GitHub picks up the PR as a linked pull request for the issue it references. I tweaked accordingly for this PR. Thanks!

kushaldas commented:
> @kushaldas Small note, can you put "Fixes #xxx" type descriptions in the _body_ of a PR, instead of the title? This makes titles a bit easier to read, and ensures that GitHub picks up the PR as a linked pull request for the issue it references. I tweaked accordingly for this PR. Thanks!

If that is required I will modify my commit messages accordingly. As those PR titles generally come from the commit messages itself, and in most projects the commit message contains which issue it is related to.

https://github.blog/2013-01-22-closing-issues-via-commit-messages/ talks about how commit message in master branch actually closes the issues.

eloquence commented:
Yes, it closes the issue, but it won't show up as a linked PR on the issue, nor will the issue show up as a linked issue on the PR. You can test that easily by removing the "Resolves" line from the PR body. So I'd recommend having it in the body to enable that cross-linking.

kushaldas commented:
> Yes, it closes the issue, but it won't show up as a linked PR on the issue, nor will the issue show up as a linked issue on the PR. You can test that easily by removing the "Resolves" line from the PR body. So I'd recommend having it in the body to enable that cross-linking.

Understood, as I said I will modify my description text to make sure that it has the issue number for cross linking from now on.

kushaldas commented:
I have updated the PR with all of the related `make clean` changes.

But, I am stuck with a different problem now, `securedrop-log` package is not getting installed on `whonix-gw-15` template.

```
    ----------
            ID: sd-whonix-install-logging
      Function: pkg.installed
        Result: None
       Comment: The following packages would be installed/updated: securedrop-log
       Started: 09:20:08.374351
      Duration: 28.308 ms
       Changes:   
  
  Summary for whonix-gw-15
  ------------
  Succeeded: 8 (unchanged=3)

```

Then, if I manually install the package, the configuration file `/etc/rsyslog.d/sdlog.conf` is not getting created in that template, or even in the `sd-whonix` vm.

I added an auditd rule to trace what is going on, I can see `dpkg` is creating a `/etc/rsyslog.d/sdlog.conf.dpkg-new` file, and doing a `SYSCALL 87` **unlink** on that file, but the actual `/etc/rsyslog.d/sdlog.conf` is never being created. 


The other problem I found (I am guessing this is upstream issue) is that when we remove a package (created via `dh-virtualenv`), the files under `/opt/venvs` are staying back on the OS.

emkll commented:
Thanks for the changes, see below for findings that should unblock you (1) as well as a regression in the `make clean` action I've observed (2). Let me know how the test plan in https://github.com/freedomofpress/securedrop-workstation/pull/447#issuecomment-585962444 goes for you.

1. I suspect `securedrop-log` isn't getting installed in whonix-ws-15 because you were not specifying highstate but rather applying the `sd-whonix-template-files.sls` file to it. Which means it doesnt configure the fpf apt repo for whonix-ws-15, as configured in `sd-workstation.top`. The diff below resolved the issue for me. 

2. One other issue in `make-clean` doesn't remove the apt-test repo, nor the apt-test key in whonix-gw-15

one last request: let's reserve squashing prior to final merge, to make debugging/review easier moving forward.
```
diff --git a/scripts/provision-all b/scripts/provision-all
index 95498f424..484a1b93d 100755
--- a/scripts/provision-all
+++ b/scripts/provision-all
@@ -18,9 +18,9 @@ echo "Set up dom0 config files, including RPC policies, and create VMs"
 sudo qubesctl --show-output state.highstate
 
 echo "Setup sd-log-buster-template vm first"
-sudo qubesctl --show-output --skip-dom0 --targets sd-log-buster-template state.sls sd-log-template-files
+sudo qubesctl --show-output --skip-dom0 --targets sd-log-buster-template state.highstate
 # Provision whonix-gw-15 with log additions because it isn't tagged with sd-workstation (we don't want it removed after a make clean)
-sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15 state.highstate sd-whonix-template-files
+sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15 state.highstate
 
 # Format list of all VMs comma-separated, for use as qubesctl target
 # We run this after dom0's highstate, so that the VMs are available for listing by tag.
```

kushaldas commented:
@emkll the branch is ready to review again, the missing part is for `make clean` to remove `gpg key` and `apt repo` from `whonix-gw-15` template.

emkll commented on scripts/provision-all:
```diff
@@ -17,6 +17,13 @@ sudo qubesctl --show-output --skip-dom0 --targets sys-firewall state.sls sd-sys-
+sudo qubesctl --show-output --skip-dom0 --targets sd-log-buster-template state.highstate
+# Provision whonix-gw-15 with log additions because it isn't tagged with sd-workstation (we don't want it removed after a make clean)
+sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15 state.highstate
+sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15 state.sls sd-whonix-template-files
```
This line is not necessary: `sd-whonix-template-files` is specified in `sd-worksation.top` so will be applied as part of the qubesctl invocation of the line directly above.

kushaldas replied:
Sadly it did not for me when I tried. I had to add that line to get it working :(

emkll replied:
Can you try again? Based on my local testing, removing L24 works. If the issue is the rebooting of `whonix-gw-15`, it may be best to `qvm-shutdown --wait sd-whonix` here intead of re-applying a salt file that has already been applied, as it will be more reliable

kushaldas replied:
`sd-whonix` or `whonix-gw-15` ?

emkll replied:
woops sorry, `whonix-gw-15`, _if_ it is required

kushaldas commented:
Now new error for me in `make all`

```
sd-proxy-buster-template:
  ----------
            ID: topd-always-passes
      Function: test.succeed_without_changes
          Name: foo
        Result: True
       Comment: Success!
       Started: 16:27:50.119208
      Duration: 0.373 ms
       Changes:   
  ----------
            ID: dsa-4371-update
      Function: cmd.script
        Result: True
       Comment: Nothing to do, apt already fixed.
       Started: 16:27:50.139560
      Duration: 621.119 ms
       Changes:   
  ----------
            ID: /etc/yum.repos.d/qubes-r4.repo
      Function: file.replace
        Result: True
       Comment: onlyif condition is false
       Started: 16:27:50.796115
      Duration: 7676.929 ms
       Changes:   
  ----------
            ID: /etc/apt/sources.list.d/qubes-r4.list
      Function: file.replace
        Result: True
       Comment: No changes needed to be made
       Started: 16:27:58.480275
      Duration: 45.276 ms
       Changes:   
  ----------
            ID: update
      Function: pkg.uptodate
        Result: True
       Comment: System is already up-to-date
       Started: 16:27:58.535297
      Duration: 10078.442 ms
       Changes:   
  ----------
            ID: install-python-apt-for-repo-config
      Function: pkg.installed
        Result: True
       Comment: All specified packages are already installed
       Started: 16:28:09.106107
      Duration: 940.491 ms
       Changes:   
  ----------
            ID: configure-apt-test-apt-repo
      Function: pkgrepo.managed
          Name: deb [arch=amd64] https://apt-test.freedom.press buster main
        Result: True
       Comment: Configured package repo 'deb [arch=amd64] https://apt-test.freedom.press buster main'
       Started: 16:28:10.049209
      Duration: 10369.041 ms
       Changes:   
                ----------
                repo:
                    deb [arch=amd64] https://apt-test.freedom.press buster main
  ----------
            ID: sd-proxy-do-not-open-here-script
      Function: file.managed
          Name: /usr/bin/do-not-open-here
        Result: True
       Comment: File /usr/bin/do-not-open-here updated
       Started: 16:28:20.447646
      Duration: 109.193 ms
       Changes:   
                ----------
                diff:
                    New file
                mode:
                    0755
  ----------
            ID: sd-proxy-do-not-open-here-desktop-file
      Function: file.managed
          Name: /usr/share/applications/do-not-open.desktop
        Result: True
       Comment: File /usr/share/applications/do-not-open.desktop updated
       Started: 16:28:20.557018
      Duration: 6.612 ms
       Changes:   
                ----------
                diff:
                    New file
                mode:
                    0644
  ----------
            ID: sd-proxy-configure-mimetypes
      Function: file.managed
          Name: /usr/share/applications/mimeapps.list
        Result: True
       Comment: File /usr/share/applications/mimeapps.list updated
       Started: 16:28:20.564299
      Duration: 5.09 ms
       Changes:   
                ----------
                diff:
                    New file
                group:
                    user
                mode:
                    0644
                user:
                    user
  ----------
            ID: sd-proxy-configure-mimetypes
      Function: cmd.run
          Name: sudo update-desktop-database /usr/share/applications
        Result: True
       Comment: Command "sudo update-desktop-database /usr/share/applications" run
       Started: 16:28:20.572030
      Duration: 367.583 ms
       Changes:   
                ----------
                pid:
                    2480
                retcode:
                    0
                stderr:
                stdout:
  ----------
            ID: install-securedrop-proxy-package
      Function: pkg.installed
        Result: False
       Comment: Problem encountered installing package(s). Additional info follows:
                
                errors:
                    - E: Unable to locate package securedrop-proxy
       Started: 16:28:20.941578
      Duration: 678.671 ms
       Changes:   
  ----------
            ID: install-securedrop-log-package
      Function: pkg.installed
        Result: False
       Comment: Problem encountered installing package(s). Additional info follows:
                
                errors:
                    - E: Unable to locate package securedrop-log
       Started: 16:28:21.620624
      Duration: 742.331 ms
       Changes:   
  ----------
            ID: install-securedrop-proxy-yaml-config
      Function: file.managed
          Name: /etc/sd-proxy.yaml
        Result: True
       Comment: File /etc/sd-proxy.yaml updated
       Started: 16:28:22.363231
      Duration: 197.961 ms
       Changes:   
                ----------
                diff:
                    New file
                mode:
                    0644
  ----------
            ID: sd-rsyslog-for-sd-proxy
      Function: file.managed
          Name: /etc/sd-rsyslog.conf
        Result: True
       Comment: File /etc/sd-rsyslog.conf updated
       Started: 16:28:22.562321
      Duration: 39.323 ms
       Changes:   
                ----------
                diff:
                    New file
                mode:
                    0644
  
  Summary for sd-proxy-buster-template
  -------------
  Succeeded: 13 (changed=7)
  Failed:     2
```

kushaldas commented:
Found the error. The network.

kushaldas commented:
- `make clean`
- `make all` on this branch
- [x] QubesIncomingLogs in `sd-log` contains `sd-app` logs
- [x] QubesIncomingLogs in `sd-log` contains `sd-proxy` logs
- [x] QubesIncomingLogs in `sd-log` contains `sd-whonix` logs
- [x] QubesIncomingLogs in `sd-log` contains `sd-devices` logs (after exporting a file to drive or print)
- [x] QubesIncomingLogs in `sd-log` contains `sd-viewer` logs (after opening a submission in a dispvm)
- [x] `make clean` works as expected
- [x] There are no RPC popups after make clean
- [ ] whonix-gw-15 no longer contains the syslog customizations (removing gpg key and apt repo is still left.

kushaldas commented:
@emkll Pushed again.

- `make clean`
- `make all` on this branch
- [x] QubesIncomingLogs in `sd-log` contains `sd-app` logs
- [x] QubesIncomingLogs in `sd-log` contains `sd-proxy` logs
- [x] QubesIncomingLogs in `sd-log` contains `sd-whonix` logs
- [x] QubesIncomingLogs in `sd-log` contains `sd-devices` logs (after exporting a file to drive or print)
- [x] QubesIncomingLogs in `sd-log` contains `sd-viewer` logs (after opening a submission in a dispvm)
- [x] `make clean` works as expected
- [x] There are no RPC popups after make clean
- [x] whonix-gw-15 no longer contains the syslog customizations

emkll commented on dom0/sd-whonix-rsyslog-enable.sls:
```diff
@@ -0,0 +1,37 @@
+    - context:
+        vmname: sd-whonix
+
+sd-rsyslog-sdlog-conf-for-sd-whonix:
```
Can we add a comment here, for future maintainers, as to why this is required here and not in other VMs, where this file is handled by the sd-log package: https://github.com/freedomofpress/securedrop-log/blob/d1799b2477f4250eae43b472472df19a49f5cb75/sdlog.conf

emkll commented on dom0/sd-clean-all.sls:
```diff
@@ -10,6 +10,17 @@ set-fedora-as-default-dispvm:
+      - qvm-run whonix-gw-15 'sudo rm -f /etc/apt/sources.list.d/securedrop_workstation.list'
+      - qvm-run whonix-gw-15 'sudo apt remove -y securedrop-log'
+      - qvm-run whonix-gw-15 'sudo systemctl restart rsyslog'
+      - qvm-run whonix-gw-15 'sudo apt-key del 4ED79CC3362D7D12837046024A3BE4A92211B03C'
```
We should also remove the prod key here:  "22245C81E3BAEB4138B36061310F561200F4AD77" (the command will always return zero so we can run both even in dev, and not worry about split logic in the cleanup action.

kushaldas replied:
Okay, i will add that too.

conorsch replied:
The `apt remove -y securedrop-log` line is not idempotent. So far, we've been careful to ensure that `make clean` exits zero even if run multiple times.

kushaldas commented:
>Templating definitely good. We might be able to use `grains["localhost"]` to get the VM name, rather than passing in `context` every time. (See output of e.g. `sudo qubesctl --skip-dom0 --show-output --target sys-firewall grains.items`.) If that works, we could stick the tasks for 1) install pkg; 2) configure rsyslog in a single state file and include it, reducing a lot of duplication. 

The whonix vms are showing the right vmname for `grains["localhost"]` or for `grains["node"]`. I am keeping the installation step same for now. If you say, then we can move out all the other log installations to a separate .sls file. But, then the `sd-log` also needs different installation (as it is the server, not the client).

emkll commented on dom0/sd-clean-all.sls:
```diff
@@ -15,9 +15,9 @@ sd-cleanup-whonix-gw-15:
     - names:
```
Is there a reason this `cmd.run` block was not moved to sd-clean-whonix.sls` ? it might make sense to centralize whonix-ws-15-specific cleanup tasks there

kushaldas replied:
Doing this.

emkll commented on dom0/sd-app-files.sls:
```diff
@@ -12,20 +12,15 @@ include:
 
 # FPF repo is setup in "securedrop-workstation" template
-install-securedrop-client-package:
+install-securedrop-client-and-securedrop-log-package:
```
:+1:

emkll commented on scripts/provision-all:
```diff
@@ -17,6 +17,13 @@ sudo qubesctl --show-output --skip-dom0 --targets sys-firewall state.sls sd-sys-
+sudo qubesctl --show-output --skip-dom0 --targets sd-log-buster-template state.highstate
+# Provision whonix-gw-15 with log additions because it isn't tagged with sd-workstation (we don't want it removed after a make clean)
+sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15 state.highstate
+#sudo qubesctl --show-output --skip-dom0 --targets whonix-gw-15 state.sls sd-whonix-template-files
```
Nit: commented out instead of deleted, might lead to confusion

conorsch commented:
Looks great. Given @emkll's thorough testing and formal approval, fine for merge. The few comments I'd given earlier are also addressed, plus we have some follow-up tickets to track (post-pilot).
